### ISSUE TITLE
UserWarning: learner.validate is only for classification problems. 

### ISSUE THREAD

**Issue by talhaanwarch** [amaiya/ktrain#77]
When i try to validate i get the following error.
```
/usr/local/lib/python3.6/dist-packages/ktrain/core.py:111: UserWarning: learner.validate is only for classification problems. For regression, etc., use learner.predict and learner.ground_truth to manually validate.
  warnings.warn('learner.validate is only for classification problems. '
```

Here is the model

```
import ktrain
from ktrain import text
MODEL_NAME = 'bert-base-uncased'
t = text.Transformer(MODEL_NAME, maxlen=100,classes=[0,1])
trn = t.preprocess_train(train, y_train)
val = t.preprocess_test(val, y_val)
model = t.get_classifier()
```

**Comment by amaiya** [comment]
I was not able to reproduce this issue with v0.10.1 of *ktrain* with either BERT or DistilBERT:

```python
# load text data
categories = ['alt.atheism', 'soc.religion.christian','comp.graphics', 'sci.med']
from sklearn.datasets import fetch_20newsgroups
train_b = fetch_20newsgroups(subset='train', categories=categories, shuffle=True)
test_b = fetch_20newsgroups(subset='test',categories=categories, shuffle=True)
(x_train, y_train) = (train_b.data, train_b.target)
(x_test, y_test) = (test_b.data, test_b.target)

# build, train, and validate model (Transformer is wrapper around transformers library)
import ktrain
from ktrain import text
MODEL_NAME = 'distilbert-base-uncased'
t = text.Transformer(MODEL_NAME, maxlen=500, class_names=train_b.target_names)
trn = t.preprocess_train(x_train, y_train)
val = t.preprocess_test(x_test, y_test)
model = t.get_classifier()
learner = ktrain.get_learner(model, train_data=trn, val_data=val, batch_size=6)
learner.fit_onecycle(5e-5, 4)
learner.validate(class_names=t.get_classes()) # class_names must be string values

# Output from learner.validate()
#                        precision    recall  f1-score   support
#
#           alt.atheism       0.92      0.93      0.93       319
#         comp.graphics       0.97      0.97      0.97       389
#               sci.med       0.97      0.95      0.96       396
#soc.religion.christian       0.96      0.96      0.96       398
#
#              accuracy                           0.96      1502
#             macro avg       0.95      0.96      0.95      1502
#          weighted avg       0.96      0.96      0.96      1502
```

The loss function for text classification problems should be one of `{'categorical_crossentropy', 'binary_crossentropy', 'sparse_categorical_crossentropy'}`, which is set automatically by `t.get_classifier()`.  If this was changed somehow, then your model might be detected as a regression model.
